frontend/src/components/GoalCards/GoalsCardsHeader.js

```diff
@@ -92,6 +104,25 @@ export default function GoalCardsHeader({
     });
   };
 
+  const onMarkSimilarGoals = async () => {
+    let similarGoals = Object.keys(allSelectedGoalIds).filter(
+      (key) => allSelectedGoalIds[key],
+    ).map((key) => parseInt(key, DECIMAL_BASE));
+
+    // If we don't just print the page.
+    if (!similarGoals.length) {
+      similarGoals = pageGoalIds;
+    }
+    // Get all the goals and associated goals from the buckets.
+    similarGoals = goalBuckets.filter(
+      (bucket) => similarGoals.includes(bucket.id),
+    ).map((bucket) => bucket.goalIds).flat();
+
+    await markSimilarGoals(recipientId, similarGoals); // PUT request to mark similar goals
+    selectAllGoalCheckboxSelect({ target: { checked: false } }); // Deselect all goals
+    history.go(0);
```

Comment:
Why do we trigger a manual reload of the page?

Reply:
do the newly added similarity group shows up in the banner

Reply:
Shouldn't this be using state to trigger re render?

Reply:
I guess I dont know what you mean

Reply:
If you have an alternative, that would work better, please suggest the code change.

Reply:
Are you reloading the page to display something that was not previously there? If yes its means you are changing the state of the page. In this case I would think you would want to use some sort of useState() var to trigger the re render and display the state change.

Reply:
Pushed suggested change.